contract/tests/test_contract.py

```diff
@@ -150,7 +150,7 @@ def setUpClass(cls):
                         0,
                         {
                             "product_id": False,
-                            "name": "Header for Services",
+                            "name": "Header for #INVOICEMONTHNAME# Services",
```

Comment:
Ok, so you should at least test the generated invoice line name.

Reply:
Hi @rousseldenis,
thank you for your review.
Sorry, but I'm a newbie with tests, I guess you told to use the marker in invoice line instead of section.
So, could be this enough to resolve?

```
cls.contract3 = cls.env["contract.contract"].create(
            {
                "name": "Test Contract 3",
                "partner_id": cls.partner.id,
                "pricelist_id": cls.partner.property_product_pricelist.id,
                "line_recurrence": False,
                "contract_type": "sale",
                "recurring_interval": 1,
                "recurring_rule_type": "monthly",
                "date_start": "2018-02-15",
                "contract_line_ids": [
                    (
                        0,
                        0,
                        {
                            "product_id": False,
                            "name": "Header for Services",
                            "display_type": "line_section",
                        },
                    ),
                    (
                        0,
                        0,
                        {
                            "product_id": False,
                            "name": "Services from #START# to #END#",
                            "quantity": 1,
                            "price_unit": 100,
                        },
                    ),
                    (
                        0,
                        0,
                        {
                            "product_id": False,
                            "name": "Line",
                            "quantity": 1,
                            "price_unit": 120,
                        },
                    ),
                    (
                        0,
                        0,
                        {
                            "product_id": False,
                            "name": "Header for #INVOICEMONTHNAME# Services",
                            "display_type": "line_section",
                        },
                    ),
                    (
                        0,
                        0,
                        {
                            "product_id": cls.product_1.id,
                            "name": "Services for #INVOICEMONTHNAME#",
                            "quantity": 1,
                            "uom_id": cls.product_1.uom_id.id,
                            "price_unit": 100,
                            "discount": 50,
                        },
                    ),
                ],
            }
```

Reply:
Hi @rousseldenis,
thank you for your review.
Sorry, but I'm a newbie with tests, could be this enough to resolve?

```
def test_check_month_name_marker(self):
    invoice_id = self.contract3.recurring_create_invoice()
    self.assertEqual(invoice_id.invoice_line_ids[0].name, "Header for May Services")
```

I'm not sure of a couple of things, first `self.contract3.recurring_create_invoice()` is called 3 times before my call: every method will increase the month so I would check "August" instead "May"?
Furthermore, I suppose test env is set to English, so can I directly set English term for the month name?

Reply:
Yes a test like should be sufficent.

But translation speaking, I'm not sure about the behaviour of the '_' function with that code:

![image](https://github.com/OCA/contract/assets/19529533/36023239-78ac-4817-8fa0-835beab9ee21)


As the translated term is not known at advance.

Either strftime provides translations (I doubt of that), either you should write a constant with a dict of every month strftime value with its translation value (which I think is the best).